pkg/statscombiner/statscombiner.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2022 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package statscombiner
+
+import (
+	"sync"
+)
+
+type statsWrapper[T any] struct {
+	stats []T
+	ttl   int
+}
+
+type StatsCombiner[T any] struct {
+	lock  sync.Mutex
+	ttl   int
+	stats map[string]*statsWrapper[T]
```

Comment:
I was wondering if we should rename the `stats` field. Maybe `statsBuffer`/`dataBuffer` or simply `buffer`? BTW, I think we should do the same [here](https://github.com/kinvolk/inspektor-gadget/blob/eea3fd22feaa689e1d2a23919bc27421a2200c21/pkg/gadgets/top/tcp/types/types.go#L87).

Reply:
What do you think about `snapshots`?

Reply:
`snapshotsWithTTL`?

And a comment:
```
	// The keys are typically the node name
```